packages/rpc-transport/src/transports/websocket/__tests__/websocket-transport-test.ts

```diff
@@ -1,54 +1,112 @@
+            });
+        }).toThrow(`'${protocol.split(':')[0]}:' is not allowed`);
+    });
+    it('throws if the URL has no protocol', () => {
+        expect(() => {
+            createWebSocketTransport({
+                sendBufferHighWatermark: Number.POSITIVE_INFINITY,
+                url: `garbage`,
+            });
+        }).toThrow("'garbage' is invalid");
+    });
+    it.each(['ws://', 'wss://', 'ws:', 'wss:', 'wS://', 'wSs://'])(
+        'does not throw if the URL begins with `%s`',
+        protocol => {
+            expect(() => {
+                createWebSocketTransport({
+                    sendBufferHighWatermark: Number.POSITIVE_INFINITY,
+                    url: `${protocol}socket`,
+                });
+            }).not.toThrow();
+        }
+    );
+});
+
+describe('IRpcWebSocketTransport', () => {
     let abortController: AbortController;
-    let transport: IRpcWebSocketTransport;
-    let ws: WS;
-    beforeEach(() => {
+    let iterator: jest.Mock;
+    let send: jest.Mock;
+    let sendWebSocketMessage: IRpcWebSocketTransport;
+    beforeEach(async () => {
         abortController = new AbortController();
-        transport = createWebSocketTransport({
+        iterator = jest.fn();
+        send = jest.fn();
+        jest.mocked(createWebSocketConnection).mockResolvedValue({
+            [Symbol.asyncIterator]: iterator,
+            send,
+        });
+        sendWebSocketMessage = createWebSocketTransport({
             sendBufferHighWatermark: Number.POSITIVE_INFINITY,
             url: 'wss://fake',
         });
-        ws = new WS('wss://fake', {
-            jsonProtocol: true,
-        });
-    });
-    afterEach(() => {
-        WS.clean();
     });
-    it('connects to the server', async () => {
+    it('creates a connection when called', () => {
         expect.assertions(1);
-        transport({ signal: abortController.signal });
-        await expect(ws.connected).resolves.toMatchObject({ readyState: WebSocket.OPEN });
+        sendWebSocketMessage({ payload: 'hello', signal: abortController.signal });
+        expect(createWebSocketConnection).toHaveBeenCalled();
     });
     it('suspends until the socket is connected', async () => {
         expect.assertions(2);
-        const transportPromise = transport({ signal: abortController.signal });
-        expect(ws.server.clients()[0]).toHaveProperty('readyState', WebSocket.CONNECTING);
-        await transportPromise;
-        expect(ws.server.clients()[0]).toHaveProperty('readyState', WebSocket.OPEN);
+        let resolveConnection: CallableFunction;
+        jest.mocked(createWebSocketConnection).mockReturnValue(
+            new Promise(r => {
+                resolveConnection = r;
+            })
+        );
+        const transportPromise = sendWebSocketMessage({ payload: 'hello', signal: abortController.signal });
+        await expect(Promise.race([transportPromise, 'pending'])).resolves.toBe('pending');
```

Comment:
This might be worth extracting into a `await expect(promise).toBePending()` Jest matcher.